### PR TITLE
Compress CAS blobs and configure output symlink capture

### PR DESCRIPTION
**Pull request by pepicrft**
## Summary
- Add zstd-backed blob storage in the local CAS while keeping BLAKE3 digests based on uncompressed bytes.
- Preserve compatibility with existing raw blobs and report logical blob sizes for compressed entries.
- Add directory-output symlink modes so script caches materialize external symlink targets by default and can opt into preserving links with `# once output-symlinks "preserve"`.
- Document the new script annotation and add tests for compression, v1 directory restore, symlink handling, and action key partitioning.

## Testing
- `mise exec -- cargo test -p once-cas -p once-core -p once-cli -p once-frontend`
- `mise exec -- cargo clippy -p once-core -p once-cli -p once-frontend --all-targets -- -D warnings`
- `mise exec -- cargo fmt --all -- --check`
- `git diff --check`

### CODE REVIEW

File: crates/once-cas/src/blob.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    let compressed = encode_all(raw, ZSTD_LEVEL)?;
+    let wrapped_len = ZSTD_BLOB_HEADER_LEN
+        .checked_add(compressed.len())
+        .ok_or_else(|| io::Error::new(io::ErrorKind::InvalidData, "blob is too large"))?;
+    if raw.starts_with(ZSTD_BLOB_MAGIC) || wrapped_len < raw.len() {
+        let raw_len = u64::try_from(raw.len())
+            .map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "blob is too large"))?;
+        let mut out = Vec::with_capacity(wrapped_len);
+        out.extend_from_slice(ZSTD_BLOB_MAGIC);
+        out.extend_from_slice(&raw_len.to_le_bytes());
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply crates/once-cas/src/blob.rs]: Fixed in 7fdd95d by moving wrapper length and raw length header checks into helpers and adding tests for checked length overflow plus u64 header overflow in crates/once-cas/src/blob.rs. Verified with `mise exec -- cargo test -p once-cas -p once-core` and `mise exec -- cargo clippy -p once-cas -p once-core --all-targets -- -D warnings`.

File: crates/once-cas/src/lib.rs [hunk 2]
```diff
@@ -132,9 +133,22 @@ impl Cas {
...
             Err(e) if e.kind() == io::ErrorKind::NotFound => Err(Error::BlobNotFound(*digest)),
+            Err(source) => Err(Error::Io {
+                path: path.clone(),
+                source,
+            }),
+        }?;
+        let mut file = File::open(&path).await.map_err(|source| Error::Io {
+            path: path.clone(),
+            source,
+        })?;
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply crates/once-cas/src/lib.rs:147]: Fixed in 7fdd95d. `Cas::blob_size` now opens the blob once, reads metadata from the same `File`, then reads the header from that handle. Verified with the once-cas and once-core tests plus clippy.

File: crates/once-core/src/directory_blob.rs [hunk 3]
```diff
@@ -1,82 +1,261 @@
...
-                metadata.permissions().mode() & 0o777
-            };
-            #[cfg(not(unix))]
-            let mode = 0o644;
-            files.push((rel, mode, std::fs::read(&path)?));
+            entries.push(capture_file_entry(&logical_path, &path, &metadata)?);
         }
     }
     Ok(())
 }
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply crates/once-core/src/directory_blob.rs:129]: Fixed in 7fdd95d. Added a short comment in `collect_symlink_entry` explaining that this `metadata` call intentionally follows the link for target classification, while the directory walk itself uses `symlink_metadata` and does not follow links.

File: crates/once-core/src/directory_blob.rs [hunk 4]
```diff
@@ -1,82 +1,261 @@
...
+            symlink_mode,
+        )?;
+        materialized_dirs.remove(&target_canonical);
+    } else if target_metadata.is_file() {
+        entries.push(capture_file_entry(
+            logical_path,
+            &resolved_target,
+            &target_metadata,
+        )?);
+    } else {
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply crates/once-core/src/directory_blob.rs:180]: Fixed in 7fdd95d. Added a comment documenting `materialized_dirs` as an active recursion stack, not a dedupe table. Repeated external directory links are intentionally materialized separately so each output path restores as a standalone subtree.

File: crates/once-core/src/action.rs [hunk 5]
```diff
@@ -8,7 +9,35 @@ use crate::{ResourceRequest, WorkspacePath};
 /// Domain-separation prefix for action digests. Bump the version when
 /// the canonical encoding (or the [`Action`] schema) changes in a way
 /// that should invalidate the cache.
-pub(crate) const ACTION_DIGEST_DOMAIN: &[u8] = b"once.action.v3\0";
+pub(crate) const ACTION_DIGEST_DOMAIN: &[u8] = b"once.action.v4\0";
+
+#[derive(Debug, Clone, Copy, Default, Serialize, Deserialize, PartialEq, Eq)]
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply crates/once-core/src/action.rs:16]: Fixed in 7fdd95d. Expanded the action digest domain comment to note that older action result JSON still deserializes through serde defaults, while the v4 domain partitions new action lookups.
